### PR TITLE
Backports to address b/336317519

### PR DESCRIPTION
**Pull request by jolivier23**
DAOS-14679 pool: Report on stopping sp_stopping (#14374)
DAOS-15514 container: fix container destroy failure (#14108)
DAOS-15672 rebuild: Fix pool destroy hangs (#14183)
DAOS-15145 pool: add pool collective function (#13764)
*DAOS-14105 object: collectively punch object (#13493)

* partial backport, just the bitmap function

Required-githooks: true
Change-Id: I2b21b8121cbdecc79ae49a464a42b1d47fb9be10

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by jolivier23** [comment]
Tested with reintegrate loop, seems to be working still.   This is the other patch of two suggested on DAOS-15386

**Comment by jolivier23** [comment]
@liw this is a bit complicated and required some manual work.  If you have cycles and could take a look, I'd appreciate it.  I was mainly trying to get your patch but the two other patches (or parts of them) seemed useful.

File: src/object/srv_obj_migrate.c [hunk 1]
```diff
@@ -3150,7 +3150,10 @@ ds_migrate_stop(struct ds_pool *pool, unsigned int version, unsigned int generat
 	uuid_copy(arg.pool_uuid, pool->sp_uuid);
 	arg.version = version;
 	arg.generation = generation;
-	rc = dss_thread_collective(migrate_fini_one_ult, &arg, 0);
+
+	rc = ds_pool_thread_collective(pool->sp_uuid,
+				       PO_COMP_ST_NEW | PO_COMP_ST_DOWN | PO_COMP_ST_DOWNOUT,
+				       migrate_fini_one_ult, &arg, 0);
```
Comment (liw) [inline src/object/srv_obj_migrate.c:3156]: [Request for changes] The "DAOS-15145 pool: add pool collective function #13764" commit is buggy, and requires at least (I can't find any other myself) the following fixes:

- DAOS-15672 rebuild: Fix pool destroy hangs (#14183)
- DAOS-15514 container: fix container destroy failure (#14108)

**Review by liw — CHANGES_REQUESTED** [review]

**Comment by liw** [comment]
> I was mainly trying to get your patch but the two other patches (or parts of them) seemed useful.

Just curious: Why do you want my humble patch? :) It doesn't seem that critical.

File: src/object/srv_obj_migrate.c (hunk 1 above)
Comment (jolivier23) [reply src/object/srv_obj_migrate.c:3156]: It was recommended by Niu as something that might help with https://daosio.atlassian.net/browse/DAOS-15386

Comment (jolivier23) [reply src/object/srv_obj_migrate.c]: sorry, this was meant as a reply to the other comment.

**Comment by jolivier23** [comment]
Bottom line

> > I was mainly trying to get your patch but the two other patches (or parts of them) seemed useful.
> 
> Just curious: Why do you want my humble patch? :) It doesn't seem that critical.

I replied to the wrong comment but ultimately, I'm just trying to take anything that might help stabilize our reintegration workflow

**Review by liw — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**